Update release script to account for mismatched git tags

Sometimes the automated release script fails with an error like `tag 'v1.4680.0' already exists`:

```
Sep 12 09:19:18 github-electron-apps app/scheduler.1299:  + git config user.email electron@github.com 
Sep 12 09:19:18 github-electron-apps app/scheduler.1299:  + git config user.name 'Electron Bot' 
Sep 12 09:19:18 github-electron-apps app/scheduler.1299:  + git add . 
Sep 12 09:19:18 github-electron-apps app/scheduler.1299:  + git commit -am 'update apps' --author 'Electron Bot <electron@github.com>' 
Sep 12 09:19:19 github-electron-apps app/scheduler.1299:  [master e08a3ce] update apps 
Sep 12 09:19:19 github-electron-apps app/scheduler.1299:  v1.4680.0 
Sep 12 09:19:19 github-electron-apps app/scheduler.1299:  npm ERR! code 128 
Sep 12 09:19:19 github-electron-apps app/scheduler.1299:  npm ERR! Command failed: git tag v1.4680.0 -am bump minor to 1.4680.0 
Sep 12 09:19:19 github-electron-apps app/scheduler.1299:  npm ERR! fatal: tag 'v1.4680.0' already exists 
Sep 12 09:19:19 github-electron-apps app/scheduler.1299:  npm ERR!  
Sep 12 09:19:19 github-electron-apps app/scheduler.1299:  npm ERR! A complete log of this run can be found in: 
Sep 12 09:19:19 github-electron-apps app/scheduler.1299:  npm ERR!     /app/.npm/_logs/2018-09-12T16_19_18_980Z-debug.log 
Sep 12 09:19:19 github-electron-apps app/scheduler.1299:  npm ERR! code ELIFECYCLE 
Sep 12 09:19:19 github-electron-apps app/scheduler.1299:  npm ERR! errno 1 
Sep 12 09:19:19 github-electron-apps app/scheduler.1299:  npm ERR! electron-apps@1.3930.0 release: `script/release.sh` 
Sep 12 09:19:19 github-electron-apps app/scheduler.1299:  npm ERR! Exit status 1
```

This is caused when the previous run of the release script successfully created a tag but didn't manage to `git push` the code with the included package.json bump.

To work around this occasional hiccup, we should catch this error and manually bump the package.json version first so the script can continue.

--- a/package.json
+++ b/package.json
@@ -15,9 +15,9 @@
     "build:pack": "node script/pack",
     "prepack": "check-for-leaks",
     "prepush": "check-for-leaks",
-    "test": "mocha --reporter min test/human-data.js test/colors-spec.js && standard --fix",
+    "test": "mocha --reporter spec test/human-data.js test/colors-spec.js && standard --fix",
     "pretest-all": "npm run build",
-    "test-all": "mocha --reporter min && standard --fix",
+    "test-all": "mocha --reporter spec && standard --fix",
     "wizard": "node wizard.js",
     "release": "script/release.sh"
   },
@@ -32,8 +32,9 @@
   "author": "Zeke Sikelianos <zeke@sikelianos.com> (http://zeke.sikelianos.com)",
   "license": "MIT",
   "devDependencies": {
+    "@octokit/rest": "^15.15.1",
     "bottleneck": "^1.16.0",
-    "chai": "^3.5.0",
+    "chai": "^4.2.0",
     "check-for-leaks": "^1.2.0",
     "cheerio": "^1.0.0-rc.2",
     "clean-deep": "^2.0.1",
@@ -42,7 +43,6 @@
     "dotenv-safe": "^4.0.4",
     "duration": "^0.2.0",
     "get-image-colors": "^1.8.1",
-    "@octokit/rest": "^15.15.1",
     "github-url-to-object": "^4.0.2",
     "human-interval": "^0.1.6",
     "husky": "^0.14.3",
@@ -59,6 +59,7 @@
     "recursive-readdir-sync": "^1.0.6",
     "rimraf": "^2.6.1",
     "sharp": "^0.20.8",
+    "sinon": "^7.2.2",
     "slugg": "^1.1.0",
     "standard": "^10.0.2",
     "yamljs": "^0.2.8"

--- a/script/release.sh
+++ b/script/release.sh
@@ -15,5 +15,6 @@ git config user.name "Electron Bot"
 git add .
 git commit -am "update apps" --author "Electron Bot <electron@github.com>"
 npm version minor -m "bump minor to %s"
-git push origin master --follow-tags
+git push origin master
+git push origin master --tags
 npm publish

--- a/test/colors-spec.js
+++ b/test/colors-spec.js
@@ -3,6 +3,7 @@
 const fs = require('fs')
 const os = require('os')
 const path = require('path')
+const sinon = require('sinon')
 
 const chai = require('chai')
 const Jimp = require('jimp')
@@ -15,45 +16,51 @@ const expect = chai.expect
 const Colors = require('../lib/colors.js')
 
 describe('colors', function () {
-  let infos
-  let errors
   let consoleInfo
   let consoleError
   let testDir
   const slugsAndIconPaths = []
 
+  const colors = ['white', 'black']
+
   before(async function () {
    // create a couple of test icons in a tmpdir
-    testDir = fs.mkdtempSync(path.join(os.tmpdir(), 'colors-spec'))
-    const colors = ['white', 'black']
+    testDir = fs.mkdtempSync(path.join(os.tmpdir(), 'colors-spec-'))
     for (const colorName of colors) {
       const c = parseInt(tinyColor(colorName).toHex8(), 16)
-      const image = await new Jimp(2, 2, c)
+      const image = new Jimp(2, 2, c)
       const iconPath = path.join(testDir, colorName + '.png')
-      image.write(iconPath)
+      await new Promise((resolve, reject) =>
+        image.write(iconPath, (err, buffer) =>
+          err ? reject(err) : resolve(buffer)
+        )
+      )
+      fs.chmodSync(iconPath, 511)
       slugsAndIconPaths.push({'slug': colorName, iconPath})
+    }
+  })
+
+  beforeEach(() => {
+    for (const colorName of colors) {
+      const iconPath = path.join(testDir, colorName + '.png')
+      fs.chmodSync(iconPath, 511)
     }
   })
 
   after(() => {
     // remove the temporaries that were created in before()
-    for (const entry of slugsAndIconPaths) { fs.unlinkSync(entry.iconPath) }
+    for (const entry of fs.readdirSync(testDir)) { fs.unlinkSync(path.resolve(testDir, entry)) }
     fs.rmdirSync(testDir)
   })
 
   beforeEach(() => {
-    errors = []
-    consoleError = console.error
-    console.error = (...args) => errors.push(args)
-
-    infos = []
-    consoleInfo = console.info
-    console.info = (...args) => infos.push(args)
+    consoleError = sinon.stub(console, 'error')
+    consoleInfo = sinon.stub(console, 'info')
   })
 
   afterEach(() => {
-    console.error = consoleError
-    console.infos = consoleInfo
+    consoleError.restore()
+    consoleInfo.restore()
   })
 
   it('should create entries with the expected properties', async () => {
@@ -79,7 +86,7 @@ describe('colors', function () {
     // newColors should be a superset of oldColors
     newColors.should.deep.contain(oldColors)
     oldColors.should.not.deep.contain(newColors)
-    expect(infos).to.have.lengthOf(2)
+    expect(consoleInfo.callCount).to.equal(2)
   })
 
   it('should remove an entry when an app is removed', async () => {
@@ -88,14 +95,14 @@ describe('colors', function () {
     // newColors should be a subset of oldColors
     newColors.should.not.deep.contain(oldColors)
     oldColors.should.deep.contain(newColors)
-    expect(infos).to.have.lengthOf(2)
+    expect(consoleInfo.callCount).to.equal(2)
   })
 
   it('should create reproducible output', async () => {
     const a = await Colors.getColors(slugsAndIconPaths, {}, testDir)
     const b = await Colors.getColors(slugsAndIconPaths, {}, testDir)
     a.should.deep.equal(b)
-    expect(infos).to.have.lengthOf(4)
+    expect(consoleInfo.callCount).to.equal(4)
   })
 
   it('should skip entries whose icons are unreadable', async() => {
@@ -104,7 +111,6 @@ describe('colors', function () {
     const input = [badEntry, goodEntry]
 
     // make the first icon unreadable
-    const oldMode = fs.statSync(badEntry.iconPath).mode
     fs.chmodSync(badEntry.iconPath, 0)
 
     const colors = await Colors.getColors(input, {}, testDir)
@@ -113,15 +119,10 @@ describe('colors', function () {
       .and
       .not.have.keys(badEntry.slug)
 
-    expect(errors)
-      .to.have.lengthOf(1)
-      .and
-      .to.satisfy(errors => JSON.stringify(errors).includes(badEntry.iconPath))
+    expect(consoleError.callCount).to.equal(1)
+    expect(consoleError.firstCall.args[0]).to.include(badEntry.iconPath)
 
-    expect(infos).to.have.lengthOf(1)
-
-    // cleanup
-    fs.chmodSync(badEntry.iconPath, oldMode)
+    expect(consoleInfo.callCount).to.equal(1)
   })
 
   it('should skip entries whose icons are unparsable', async() => {
@@ -137,15 +138,10 @@ describe('colors', function () {
       .and
       .not.have.keys(badEntry.slug)
 
-    expect(errors)
-      .to.have.lengthOf(1)
-      .and
-      .to.satisfy(errors => JSON.stringify(errors).includes(badEntry.iconPath))
+    expect(consoleError.callCount).to.equal(1)
+    expect(consoleError.firstCall.args[0]).to.include(badEntry.iconPath)
 
-    expect(infos).to.have.lengthOf(2)
-
-    // cleanup
-    fs.unlinkSync(badEntry.iconPath)
+    expect(consoleInfo.callCount).to.equal(2)
   })
 
   it('should update revHashes when icon files change', async() => {
@@ -169,6 +165,6 @@ describe('colors', function () {
       .property('revHash')
       .should.not.equal(oldColors[changedEntry.slug].source.revHash)
 
-    expect(infos).to.have.lengthOf(3)
+    expect(consoleInfo.callCount).to.equal(3)
   })
 })

